lanelet2_io/src/OsmFile.cpp

```diff
@@ -95,12 +95,30 @@ void removeAndFixPlaceholders(Primitive** toRemove, Roles& fromRoles, std::vecto
 
 class OsmFileWriter {
  public:
-  static std::unique_ptr<pugi::xml_document> write(const File& osmFile) {
+  static std::unique_ptr<pugi::xml_document> write(const File& osmFile, const io::Configuration& params) {
     auto xml = std::make_unique<pugi::xml_document>();
     auto osmNode = xml->append_child(keyword::Osm);
     osmNode.append_attribute("version") = "0.6";
+    // whether to upload the model in JOSM, true by default
+    {
+      const auto iter = params.find("josm_upload");
+      if (iter != params.end() && !iter->second.asBool().value_or(true)) {
```

Comment:
Changing the default to `upload="false"` was how I had this originally in this PR. However, we decided that we should have a parameter and therefore we can keep the previous default for backwards compatibility. I don't want to introduce any breaking changes.

Reply:
Well this was mostly a about the precision of the elevation. Before, we just didn't set `upload` at all, so someone relying on the exact shape of the output (I hope not) will get a problem in any case.

I'm just thinking that if for the majority of users `upload=false` is the better choice, I'm in favor of making it the default, even at the risk of breaking backwards compability for a few that really relied on the old behavior.

Reply:
You are correct. I double checked it and on master there's no `osmNode.append_attribute("upload") = "true";` statement at all! I think what happens is after you edit in JOSM and save the file the attribute `upload="true"` gets added unless there's already the `upload="false"` in the file to prevent that. Ok, I switched to `upload="false"` by default (which is how I originally had it in the first version of this PR).